pkg/apis/configuration/v1alpha1/kong_license_types.go

```diff
@@ -0,0 +1,107 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Namespaced,shortName=kl,categories=kong-ingress-controller,path=konglicenses
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+
+// KongLicense stores a Kong enterprise license to apply to managed Kong gateway instances.
+type KongLicense struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// RawLicenseString is the raw content of the license in string format.
+	RawLicenseString string `json:"rawLicenseString"`
+	// Enabled is set to true to let controllers (like KIC) to reconcile it.
+	// Default value is true to apply the license by default.
+	// +kubebuilder:default=true
+	Enabled bool `json:"enabled"`
+	// Status is the status of the KongLicense being processed by controllers.
+	Status KongLicenseStatus `json:"status,omitempty"`
+}
+
+// KongLicenseStatus stores the status of the KongLicense being processesed in each controller that reconciles it.
+type KongLicenseStatus struct {
+	// +listType=map
+	// +listMapKey=controllerName
+	KongLicenseControllerStatuses []KongLicenseControllerStatus `json:"controllers,omitempty"`
+}
+
+// KongLicenseControllerStatus is the status of owning KongLicense being processed
+// identified by the controllerName field.
+type KongLicenseControllerStatus struct {
+	// ControllerName is an identifier of the controller to reconcile this KongLicense.
+	// Should be unique in the list of controller statuses.
+	ControllerName string `json:"controllerName"`
+	// ControllerRef is the reference of the controller to reconcile this KongLicense.
+	// It is usually the name of (KIC/KGO) pod that reconciles it.
+	ControllerRef *ControllerReference `json:"controllerRef,omitempty"`
+	// Configured is set to true if the controller applied the content of the license on managed Kong gateway.
+	Configured bool `json:"configured"`
```

Comment:
```suggestion
```

I'd omit that similarly to `Phase` as that can be expressed in the `Programmed` condition.

Reply:
Removed it.